baseservice/iodevices/file_system/file_system_input_device.py

```diff
@@ -40,38 +42,50 @@ def __init__(self, device: 'FileSystemInputDevice', org_path: str, tmp_path: str
 
     @property
     def org_path(self) -> str:
+        """
+        the original path of the file in this transaction
+        """
         return self._org_path
 
     @property
     def tmp_path(self) -> str:
+        """
+        the temp (current) path of the file in this transaction
+        """
         return self._tmp_path
 
     @staticmethod
-    def _get_backout_filename(org_filename: str) -> str:
+    def _get_poison_filename(org_filename: str) -> str:
         """
-        gets a filename for backout
+        gets a filename for poison messages
 
         :param org_filename: the original filename
-        :return: filepath for backout
+        :return: filepath for the poison message
         """
         basename = os.path.basename(org_filename)
         dirname = os.path.dirname(org_filename)
-        backout_folder = os.path.join(dirname, 'BACKOUT')
-        create_dir_if_not_exists(backout_folder)
-        backout_filename = os.path.join(backout_folder, get_random_id() + basename)
-        return backout_filename
+        poison_folder = os.path.join(dirname, 'POISON')  # TODO: get this from the user
+        create_dir_if_not_exists(poison_folder)
```

Comment:
Question, why are we against always creating the poison folder and just let it be empty?
Is it because you want to keep these stuff out of the user's view? You already have the book keeping folder by default, so why not put both of these folders in some sort of a single `.internal` folder (like `.git` does)? That way the user will still see exactly one folder that they shouldn't care about, you will have a consistent folder structure, and `_get_poision_filename` will not have this unexpected side-effect.

Reply:
notice that it's a POISON folder under each queue.,.. not a single poison folder.